lib/wickedbase.pm

```diff
@@ -379,18 +379,21 @@ sub upload_wicked_logs {
     my $dir_name = $self->{name} . '_' . $prefix;
     my $logs_dir = "/tmp/$dir_name";
     record_info('Logs', "Collecting logs in $logs_dir");
-    script_run("mkdir -p $logs_dir");
-    script_run("date +'%Y-%m-%d %T.%6N' > $logs_dir/date");
-    script_run("journalctl -b -o short-precise|tail -n +2 > $logs_dir/journalctl.log");
-    script_run("wicked ifstatus --verbose all > $logs_dir/wicked_ifstatus.log 2>&1");
-    script_run("wicked show-config > $logs_dir/wicked_config.log 2>&1");
-    script_run("wicked show-xml > $logs_dir/wicked_xml.log 2>&1");
-    script_run("ip addr show > $logs_dir/ip_addr.log 2>&1");
-    script_run("ip route show table all > $logs_dir/ip_route.log 2>&1");
-    script_run("cat /etc/resolv.conf > $logs_dir/resolv.conf 2>&1");
-    script_run("cp /tmp/wicked_serial.log $logs_dir/") if $prefix eq 'post';
-    script_run("tar -C /tmp/ -cvzf $dir_name.tar.gz $dir_name");
+    my $result = eval {
+        script_run("mkdir -p $logs_dir");
+        script_run("date +'%Y-%m-%d %T.%6N' > $logs_dir/date");
+        script_run("journalctl -b -o short-precise|tail -n +2 > $logs_dir/journalctl.log");
+        script_run("wicked ifstatus --verbose all > $logs_dir/wicked_ifstatus.log 2>&1");
+        script_run("wicked show-config > $logs_dir/wicked_config.log 2>&1");
+        script_run("wicked show-xml > $logs_dir/wicked_xml.log 2>&1");
+        script_run("ip addr show > $logs_dir/ip_addr.log 2>&1");
+        script_run("ip route show table all > $logs_dir/ip_route.log 2>&1");
+        script_run("cat /etc/resolv.conf > $logs_dir/resolv.conf 2>&1");
+        script_run("cp /tmp/wicked_serial.log $logs_dir/") if $prefix eq 'post';
+        script_run("tar -C /tmp/ -cvzf $dir_name.tar.gz $dir_name");
+    };
```

Comment:
script_run() should not raise an exception or?

Reply:
is it ok now ?

Reply:
yes